app/components/agenda/agenda-header/agenda-actions.js

```diff
@@ -352,6 +353,43 @@ export default class AgendaAgendaHeaderAgendaActions extends Component {
     this.router.refresh(this.router.currentRouteName);
   }
 
+  get canEmptyInternalReviews() {
+    // action will do nothing on designAgenda A, so hide it instead
+    const isDesignAgendaA = this.args.currentAgenda.status.get('isDesignAgenda') && this.args.currentAgenda.serialnumber === 'A';
+    return this.currentSession.may('manage-agendaitems') && !isDesignAgendaA;
+  }
+
+  emptyInteralReviews = async() => {
+    this.showConfirmEmptyInternalReviews = false;
+    this.args.onStartLoading(this.intl.t('empty-internal-review'));
+    // getting all valid agendaitems first gets better results than trying submission-internal-review directly via subcase
+    const approvedAgendaitems = await this.store.queryAll('agendaitem', {
+      'filter[:has:previous-version]': true,
+      'filter[agenda][:id:]': this.args.currentAgenda.id,
+    });
+    const savePromises = approvedAgendaitems.map((internalReview) => this.emptyInteralReviewsOfAgendaitemThrottled.perform(internalReview));
+    await all(savePromises);
+    // TODO KAS-4886 this can go when we no longer have to save agendaitems
+    this.args.onStopLoading();
+    this.args.didApproveAgendaitems();
+  };
+
+  emptyInteralReviewsOfAgendaitemThrottled = task({ maxConcurrency: 5}, async (agendaitem) => {
+    const internalReview = await this.store.queryOne('submission-internal-review', {
+      'filter[subcase][agenda-activities][agendaitems][:id:]': agendaitem.id,
+    })
+    if (internalReview?.id && !isEmpty(internalReview.privateComment)) {
+      internalReview.privateComment = '';
+      return await internalReview.save();
+    }
+    // This property is still filled in as of now, we should empty it to avoid showing this briefly in agenda overview
+    // TODO KAS-4886 remove when property is removed from model
+    if (!isEmpty(agendaitem.privateComment)) {
+      agendaitem.privateComment = '';
+      return await agendaitem.save();
```

Comment:
Could have some unintended effects on approved agendas that have a next version. Will need to implement an extra check here, or wait until https://kanselarij.atlassian.net/browse/KAS-4886 is done.

Reply:
fixed this by passing `reverseSortedAgendas` from args. We can use that to determine the latest agenda without awaiting. Improved check to also only allow the action on the latestAgenda.